src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/MandateForChange.kt

```diff
@@ -13,27 +13,29 @@ class MandateForChange(
 ) {
   fun hasNoMandate(crn: String, convictions: Collection<Conviction>): Boolean =
     convictions
-      .filter { it.sentence?.terminationDate == null }
-      .let { activeConvictions ->
-        activeConvictions.none {
-          it.sentence != null && (isCustodial(it.sentence) || hasNonRestrictiveRequirements(crn, it.convictionId))
-        }
+      .filter { currentSentence(it.sentence) }
+      .none {
+        isCustodial(it.sentence) || hasNonRestrictiveRequirements(crn, it.convictionId)
```

Comment:
nitpick but this could be 
`.none {
     currentSentence(it.sentence) && ( isCustodial(it.sentence) || hasNonRestrictiveRequirements(crn, it.convictionId))`

Not sure what is more readable.